test/tests.py

```diff
@@ -1508,6 +1509,154 @@ def test(self):
+                curr_func = None
+
+        assert reset, "Line directive not reset by end of C file"
+        assert curr_func is None, "Unterminated C function"
+
+
+    def test(self):
+        super().test()
+        c_file = Path(f'{self.cfilename}.c').resolve()
+        h_file = Path(f'{self.cfilename}.h').resolve()
+        base_file = Path(self.filename).resolve()
+
+        func_spans = {}
+        cblock_spans = {}
+        not_linemarked = set()
+        with open(base_file) as base_file_handle:
+            curr_span = None
+            for (lineno, line) in enumerate(base_file_handle, 1):
+                line = line.strip()
+                m = begin_linemarks_func_re.match(line)
+                if m:
+                    assert curr_span is None, f"{lineno}: {curr_span}"
+                    curr_span = (func_spans, m.group(1).split(), lineno + 1)
+                elif line == '// BEGIN-LINEMARKS-CBLOCK':
+                    curr_span = (cblock_spans, lineno + 1)
+                elif line == '// END-LINEMARKS':
+                    assert curr_span is not None
+                    if curr_span[0] is func_spans:
+                        for func in curr_span[1]:
+                            func_spans.setdefault(func, set()).add(
+                                (curr_span[2], lineno - 1))
+                    else:
+                        cblock_spans[curr_span[1]] = lineno - 1
+                    curr_span = None
+                elif line == '// NOT-LINEMARKED':
+                    assert (curr_span is not None
+                            and curr_span[0] is func_spans), \
+                            ("{lineno}: // NOT-LINEMARKED used outside of "
+                             + "LINEMARKS-FUNC block")
+                    not_linemarked.add(lineno + 1)
+                if (curr_span is not None and curr_span[0] is func_spans
+                    and (not line or line.startswith('//'))):
+                    not_linemarked.add(lineno)
+            assert curr_span is None
+
+        expected_func_lines = {
+            name: { line: False
+                   for (start, end) in ranges
+                   for line in range(start, end + 1)
+                   if line not in not_linemarked }
+            for (name, ranges) in func_spans.items()}
+        found_cblocks = set()
+        for curr_file in (c_file, h_file):
+            with open(curr_file) as file_handle:
+                curr_cblock_expected = []
+                for (dml_lineno, c_lineno, curr_func) in self.dml_lines_iter(
+                        curr_file, file_handle.readlines()):
+                    if curr_cblock_expected:
+                        (expected_dml_lineno,
+                         expected_c_lineno) = curr_cblock_expected.pop()
+                        assert ((dml_lineno, c_lineno)
+                                == (expected_dml_lineno, expected_c_lineno)), \
+                                (f'In {curr_file}: Expected DML line '
```

Comment:
I think the style guide wants us to avoid `\` line continuations, which *can* be done by movingthe `(` of the message on the previous line. This stuff is dangerous though, it's way too easy to accidentally write `assert (x, y)` instead of `assert x, y`. I don't know how to make this easy to parse, perhaps something like this?
```
assert (
    ...
), (
    ...
)
```

Reply:
errgh... maybe.

I hate Python's breed of syntactically significant whitespace.